pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go

```diff
@@ -36,21 +37,36 @@ var (
 	BalancedResourceAllocationMap = balancedResourcePriority.PriorityMap
 )
 
-func balancedResourceScorer(requested, allocable *schedulercache.Resource) int64 {
+func balancedResourceScorer(requested, allocable *schedulercache.Resource, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
 	cpuFraction := fractionOfCapacity(requested.MilliCPU, allocable.MilliCPU)
 	memoryFraction := fractionOfCapacity(requested.Memory, allocable.Memory)
+	// This to find a node which has most balanced CPU, memory and volume usage.
+	if includeVolumes && utilfeature.DefaultFeatureGate.Enabled(features.VolumesOnNodeForBalancing) && allocatableVolumes > 0 {
+		volumeFraction := float64(requestedVolumes) / float64(allocatableVolumes)
+		if cpuFraction >= 1 || memoryFraction >= 1 || volumeFraction >= 1 {
+			// if requested >= capacity, the corresponding host should never be preferred.
+			return 0
+		}
+		// Compute variance for all the three fractions.
+		mean := (cpuFraction + memoryFraction + volumeFraction) / float64(3)
+		variance := float64(((cpuFraction - mean) * (cpuFraction - mean)) + ((memoryFraction - mean) * (memoryFraction - mean)) + ((volumeFraction - mean) * (volumeFraction - mean)))
```

Comment:
variance could be larger than 1, causing `1 - variance` (below) to be negative. I guess you have forgotten to divide by 3.

Reply:
> I guess you have forgotten to divide by 3.

:rofl:  Lol, thanks for catching it :)

> variance could be larger than 1, causing 1 - variance (below) to be negative.

I thought about that, but it seems our sample set always has fractions and 
- fraction-mean(which is a fraction) would be fraction again.
- square-root of (fraction-mean) will be further lower than original fraction

 Not sure if I missed any edge case.

Reply:
You are right, but given that you have not divided by 3, it could be larger than 1. For example, if each component is 0.9, then variance will be 0.81 + 0.81 + 0.81. If you divide by 3, variance will be less than 1.

Reply:
So, you were commenting on my version of variance(without division by 3) I thought you were commenting on variance in general that it could be greater than 1(which is true as well). Sure, I will make that change.

Reply:
Yes, I was referring to this implementation.